flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BigQueryBufferedWriter.java

```diff
@@ -225,6 +266,10 @@ public Collection<BigQueryCommittable> prepareCommit()
     @Override
     public List<BigQueryWriterState> snapshotState(long checkpointId) {
         logger.info("Snapshotting state in subtask {} for checkpoint {}", subtaskId, checkpointId);
+        // Since we are moving towards checkpointing and write() for previous checkpoint is
+        // completed,
+        // reset the flag isFirstWriteAfterCommit
+        isFirstWriteAfterCheckpoint = true;
```

Comment:
Lets be consistent with referring to the variable as `isFirstWriteAfterCheckpoint` or `isFirstWriteAfterCommit`

Reply:
Ack, Fixed.